### PR TITLE
[MaterializedView]Drop Materialized vied stmt

### LINKED ISSUE #4781: [MaterializedView] modify DropMaterializedViewStmt for MaterializedView[P0]
## Feature request

**Is your feature request related to a problem? Please describe.**

to #3680

**Describe the solution you'd like**

add analyze api for new MaterializedView

### PR DESCRIPTION
**Pull request by zddr**
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] others

## Which issues of this PR fixes ：

Fixes #4781

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 0 / 37 (00.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|0|2|00.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|0|8|00.00%|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/analysis/DropMaterializedViewStmt.java|0|3|00.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java|0|23|00.00%|

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocksLex.g4 [hunk 1]
```diff
@@ -183,6 +183,7 @@ WHEN: 'WHEN';
 WHERE: 'WHERE';
 WITH: 'WITH';
 YEAR: 'YEAR';
+MATERIALIZED: 'MATERIALIZED';
```
Comment (HangyuanLiu) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocksLex.g4:186]: MATERIALIZED is noReserved

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4 [hunk 2]
```diff
@@ -29,6 +29,7 @@ statement
     | SHOW FULL? TABLES ((FROM | IN) db=qualifiedName)?
         ((LIKE pattern=string) | (WHERE expression))?                                   #showTables
     | SHOW DATABASES ((LIKE pattern=string) | (WHERE expression))?                      #showDatabases
+    | DROP MATERIALIZED VIEW (IF EXISTS)?mvName=qualifiedName (ON tableName = qualifiedName)?              #dropMaterialized
```
Comment (HangyuanLiu) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4:32]: why "on" keyword used here?

**Review by HangyuanLiu — COMMENTED** [review]
Please add ut cover

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4 [hunk 3]
```diff
@@ -29,6 +29,7 @@ statement
     | SHOW FULL? TABLES ((FROM | IN) db=qualifiedName)?
         ((LIKE pattern=string) | (WHERE expression))?                                   #showTables
     | SHOW DATABASES ((LIKE pattern=string) | (WHERE expression))?                      #showDatabases
+    | DROP MATERIALIZED VIEW (IF EXISTS)?mvName=qualifiedName              #dropMaterialized
```
Comment (HangyuanLiu) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4:32]:
```suggestion
    | DROP MATERIALIZED VIEW (IF EXISTS)? mvName=qualifiedName              #dropMaterialized
```

**Review by HangyuanLiu — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 18 / 27 (66.67%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|0|4|00.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java|5|8|62.50%|
|:large_blue_circle:|com/starrocks/analysis/DropMaterializedViewStmt.java|8|9|88.89%|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|3|3|100.00%|

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+    static class MaterializedViewAnalyzerVisitor extends AstVisitor<Void, ConnectContext> {
+        public void analyze(StatementBase statement, ConnectContext session) {
+            visit(statement, session);
+        }
+
+        @Override
+        public Void visitDropMaterializedViewStatement(DropMaterializedViewStmt stmt, ConnectContext context) {
+            stmt.getDbMvName().normalization(context);
+            return null;
```
Comment (ABingHuang) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java:23]: why not use void?

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 21 / 25 (84.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|3|4|75.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java|5|6|83.33%|
|:large_blue_circle:|com/starrocks/analysis/DropMaterializedViewStmt.java|8|9|88.89%|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|3|3|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 21 / 25 (84.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|3|4|75.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java|5|6|83.33%|
|:large_blue_circle:|com/starrocks/analysis/DropMaterializedViewStmt.java|8|9|88.89%|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|3|3|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 26 / 30 (86.67%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|3|4|75.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java|5|6|83.33%|
|:large_blue_circle:|com/starrocks/analysis/DropMaterializedViewStmt.java|8|9|88.89%|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/sql/StatementPlanner.java|1|1|100.00%|

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/Analyzer.java [hunk 5]
```diff
@@ -134,6 +135,12 @@ public Void visitDeleteStatement(DeleteStmt node, ConnectContext context) {
             DeleteAnalyzer.analyze(node, context);
             return null;
         }
+
+        @Override
+        public Void visitDropMaterializedViewStatement(DropMaterializedViewStmt statement, ConnectContext context) {
```
Comment (ABingHuang) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/Analyzer.java:140]: can add enableNewMaterializedView here?
if. enableNewMaterializedView is false, throw an exception here?

**Review by HangyuanLiu — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 21 / 25 (84.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|3|4|75.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java|5|6|83.33%|
|:large_blue_circle:|com/starrocks/analysis/DropMaterializedViewStmt.java|8|9|88.89%|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|3|3|100.00%|

**Review by Astralidea — APPROVED** [review]

**Pull request merged**